### PR TITLE
Add logging configuration to django

### PR DESCRIPTION
**Pull request by bartoszbetka**

### CODE REVIEW

File: project_liberation/settings/base.py [hunk 1]
```diff
@@ -143,6 +144,99 @@
...
+            # a lot of disk space.
+            'handlers':  ['console'],
+            # NOTE: Changing level of this logger will change levels of loggers from plugins
+            # because they often don't have a level set explicitly and inherit this one instead.
+            'level':     'DEBUG',
+            'propagate': False,
+        },
+        'py.warnings': {
+            # Prevent Python from printing its warnings to the console. Our top-level logger already handles and prints them.
+            # I'm not entirely sure why this works but I think that the default py.warnings has a custom console handler
```
Comment (pawelkisielewicz) [inline project_liberation/settings/base.py]: Do you really do this against rules you wrote above (please see other comments) and write here that you do it and don't know why? Why?

File: project_liberation/settings/base.py [hunk 2]
```diff
@@ -143,6 +144,99 @@
...
+        # - If a message is logged to logger 'a.b' and propagates to logger 'a', only the level of logger
+        #   'a.b' counts. Level of logger 'a' is ignored. Not kidding. See the diagram:
+        #   https://docs.python.org/3/howto/logging.html#logging-flow
+        # - level of logger 'a.b' determines not only what it handles but also what it propagates to parent.
+        # - If a logger has no level, it inherits level from parent. Root logger (the one called '') has level WARNING by default.
+
+        # RULES: Try to stick to the following conventions:
+        # - Don't set level of a logger unless you explicitly want to prevent some messages from being handled or propagated.
+        # - In most cases it's better to leave level at DEBUG here and set level in handler instead.
+        # - Set level explicitly if you don't propagate. Such a logger should not be dependent on parent's level.
```
[the remark this answers is not in the document]
Comment (cameel) [reply project_liberation/settings/base.py]: No. It says that the logger dict must contain a key called `'level'` key with non-empty value when it contains `'propagate': False`. The rule does not say anything about the case when `'propagate'` is `True` or missing.

And as you can see below, the only logger with `'propagate' : False` is `''` and it does have `'level': 'DEBUG'` defined explicitly. I don't see why you think it breaks the rule.

File: project_liberation/settings/base.py [hunk 3]
```diff
@@ -143,6 +144,99 @@
...
+    'loggers': {
+        # NOTE: There are a few important caveats you need to consider when tweaking logging:
+        # - If a message is logged to logger 'a.b' and propagates to logger 'a', only the level of logger
+        #   'a.b' counts. Level of logger 'a' is ignored. Not kidding. See the diagram:
+        #   https://docs.python.org/3/howto/logging.html#logging-flow
+        # - level of logger 'a.b' determines not only what it handles but also what it propagates to parent.
+        # - If a logger has no level, it inherits level from parent. Root logger (the one called '') has level WARNING by default.
+
+        # RULES: Try to stick to the following conventions:
+        # - Don't set level of a logger unless you explicitly want to prevent some messages from being handled or propagated.
```
[the remark this answers is not in the document]
Comment (cameel) [reply project_liberation/settings/base.py]: It's fine, the rule is just not 100% precise because I did not want to state the obvious and make it more confusing than it needs to be.

If you want this to be precise you could reword it to say `Don't set level of a logger to anything higher than DEBUG ...` and clarify that setting `DEBUG` is also fine when a different level would be inherited from the parent logger. I didn't think it was necessary.

File: project_liberation/settings/base.py (hunk 1 above)
Comment (cameel) [reply project_liberation/settings/base.py]: I know what it does, and I checked that it works (already quite a long time ago). Adding this rule makes python print warnings only once to the console which is what we want. Without this setting `DEBUG` for `''` causes them to be printed twice.

The only thing I don't really know here is why it has this effect. I have an idea what it might be doing but I never had a time to verify my suspicions and it was not all that relevant. Logging has quite a few quirks in various corner cases and even bugs and I didn't really feel it was necessary to all the way down to the implementation to find out. Feel free to dig deeper into this if you're curious why it does this.
